core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -182,6 +182,7 @@ public static Status httpStatusToGrpcStatus(int httpStatusCode) {
     }
     switch (httpStatusCode) {
       case HttpURLConnection.HTTP_BAD_REQUEST:  // 400
+      case 431: // Request Header Fields Too Large
```

Comment:
Header processing fails in NettyClientTransport test otherwise I think.  The change is not made there because this preserves existing error behavior (INTERNAL instead of UNKNOWN).